src/main/scala/cognite/spark/v1/SyncCursorCallback.scala

```diff
@@ -0,0 +1,60 @@
+package cognite.spark.v1
+
+import cats.effect.IO
+import com.cognite.sdk.scala.sttp.{GzipBackend, RetryingBackend}
+import io.circe.Codec
+import io.circe.generic.semiauto.deriveCodec
+import scala.concurrent.duration._
+import sttp.client3.{SttpBackend, basicRequest}
+import sttp.client3.asynchttpclient.SttpClientBackendFactory
+import sttp.client3.asynchttpclient.cats.AsyncHttpClientCatsBackend
+import sttp.client3.circe.{asJson, circeBodySerializer}
+import sttp.model.{MediaType, Uri}
+
+case class IncrementalCursor(name: String, value: String)
+case class IncrementalCursorResponse(name: String, value: String, updated: Boolean)
+
+object SyncCursorCallback {
+  @transient private lazy val sttpBackend: SttpBackend[IO, Any] =
+    new GzipBackend[IO, Any](
+      AsyncHttpClientCatsBackend.usingClient(SttpClientBackendFactory.create("Last-Cursor-Submitter")))
+
+  @transient private val retryingBackend = new RetryingBackend[IO, Any](
+    sttpBackend,
+    maxRetries = 5,
+    initialRetryDelay = 200.milliseconds,
+    maxRetryDelay = 5.seconds
+  )
+
+  implicit val incrementalCursorCodec: Codec[IncrementalCursor] = deriveCodec[IncrementalCursor]
+  implicit val incrementalCursorResponseCodec: Codec[IncrementalCursorResponse] =
+    deriveCodec[IncrementalCursorResponse]
+
+  def lastCursorCallback(
+      callbackUrl: String,
+      cursorName: String,
+      cursorValue: String,
+      jobId: String): IO[IncrementalCursorResponse] = {
+    val uri = Uri.parse(s"$callbackUrl/$jobId") match {
+      case Left(value) => throw new IllegalArgumentException(s"Failed to parse URI '$value'")
+      case Right(value) => value
+    }
+
+    basicRequest
+      .followRedirects(false)
+      .contentType(MediaType.ApplicationJson)
+      .acceptEncoding(MediaType.ApplicationJson.toString())
+      .body(IncrementalCursor(cursorName, cursorValue))
+      .post(uri)
+      .response(asJson[IncrementalCursorResponse])
+      .mapResponse {
```

Comment:
a little unsure of the ordering or layering between `.mapResponse` and `RetryingBackend.send` so yeah, should be simpler to ignore errors on caller level